### ISSUE TITLE
[Feature] Laravel Dusk tests

### ISSUE THREAD

**Issue by bbashy** [REBELinBLUE/deployer#336]
Would you be willing to add Laravel Dusk tests to this project?

I've written a good number of tests on some projects and I know you love testing so hopefully I can get the ball rolling (pun intended).

![](https://d.pr/KnZ8/3qde1jAX.jpg)

**Comment by REBELinBLUE** [comment]
Absolutely, you'll notice the Browser folder exists in tests/, I just didn't want to look at it until I start on rewriting the horrendous javascript 😂

**Comment by bbashy** [comment]
That's a good point. Were you going to redo some front-end elements as well?

I can start writing some basic ones just for forms if you wanted? Maybe some folder structure ideas.

**Comment by REBELinBLUE** [comment]
I hadn't planned to change the UI in any way

**Comment by bbashy** [comment]
JavaScript shouldn't modify the elements a lot then?

**Comment by REBELinBLUE** [comment]
Fair point, it shouldn't

**Comment by cerw** [comment]
Isn't this more for phptesting.org ?

**Comment by bbashy** [comment]
@cerw You're thinking about it the other way around I think? This issue is for adding actual dusk tests to deployer repo, not deployment.

**Comment by cerw** [comment]
@bbashy sorry my bad. I just implemented my dusk tests on https://github.com/block8/phpci , its tricky one, be great to know how guys implementing it on CI
